environmental_data_modules/aurn_imputation_tests.py

```diff
@@ -428,20 +517,33 @@ def imputation_hourly_analysis(self,hourly_imputed_dataframe,hourly_reference_da
                     data_imputed   = data_imputed[data_reference.notna()] 
                     data_reference = data_reference[data_reference.notna()] 
                 
-                    # calculate the Mean Square Error
-                    #mserror = mse(data_reference,data_imputed)
-                
                     # plot scatter
                     data_combined = pd.DataFrame()
                     data_combined[spc] = data_reference
                     data_combined['{} (imputed)'.format(spc)] = data_imputed
+
+                    min_val = min(min(data_imputed),min(data_reference))
+                    max_val = max(max(data_imputed),max(data_reference))
+                    range_val = max_val - min_val
+                    
+                    k_corr, k_pval = stats.kendalltau(data_reference,data_imputed)
+                    s_corr, s_pval = stats.spearmanr(data_reference,data_imputed)
+                    p_corr, p_pval = stats.pearsonr(data_reference,data_imputed)
+                    slope, intercept, r_value, p_value, std_err = stats.linregress(data_reference,data_imputed)
+                    hourly_stat_dataset.loc[(site,spc),col_headers] = [k_corr,s_corr,p_corr,slope,r_value**2,p_value,std_err]
                 
-                    sns_plot = sns.jointplot(data=data_combined,x=spc,y='{} (imputed)'.format(spc))
+                    sns_plot = sns.jointplot(data=data_combined,x=spc,y='{} (imputed)'.format(spc),kind="reg")
+                    sns_plot.ax_joint.plot(data_combined['{} (imputed)'.format(spc)],data_combined['{} (imputed)'.format(spc)], 'r-', linewidth=1)
+                    sns_plot.ax_joint.set_xlim(min_val-range_val*0.05,max_val+range_val*0.05)
+                    sns_plot.ax_joint.set_ylim(min_val-range_val*0.05,max_val+range_val*0.05)
+                    sns_plot.ax_joint.text(min_val+range_val*0.1,max_val-range_val*0.1,'KendallTau; corr = {0:.2f}; p = {1:.2f}'.format(k_corr,k_pval))
+
                     pdf_pages.savefig(sns_plot.fig)
                     plt.close()
                     
         
-        #print('...analysis of hourly imputed data is work in progress...')
+        hourly_stat_dataset.to_csv('aurn_hourly_correlation_stats.csv', index=True, header=True, float_format='%.4f')
```

Comment:
Could outfile name and float_format be a static/constant? E.g see AurnExtractor and MetExtractor